types/msg.go

```diff
@@ -96,6 +100,43 @@ type IBCMsg struct {
 	CloseChannel *CloseChannelMsg `json:"close_channel,omitempty"`
 }
 
+type GovMsg struct {
+	// This maps directly to [MsgVote](https://github.com/cosmos/cosmos-sdk/blob/v0.42.5/proto/cosmos/gov/v1beta1/tx.proto#L46-L56) in the Cosmos SDK with voter set to the contract address.
+	Vote VoteMsg `json:"vote,omitempty"`
+}
+
+type VoteOption int
+
+type VoteMsg struct {
+	ProposalId uint64     `json:"proposal_id"`
+	Vote       VoteOption `json:"vote"`
+}
+
+const (
+	Yes VoteOption = iota
+	No
+	Abstain
+	NoWithVeto
+)
+
+var toVoteOption = map[string]VoteOption{
+	"yes":          Yes,
+	"no":           No,
+	"abstain":      Abstain,
+	"no_with_veto": NoWithVeto,
+}
+
+func (s *VoteOption) UnmarshalJSON(b []byte) error {
+	var j string
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return err
+	}
+	// Note that if the string cannot be found then it will be set to the zero value, 'Created' in this case.
```

Comment:
'Created'? Where does this come from?

Reply:
Oops.

I found this pattern here: https://gist.github.com/lummie/7f5c237a17853c031a57277371528e87

Reply:
What this code does not show you: how to handle errors. There is a second return value: https://tour.golang.org/moretypes/22